pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/ValidationManager.java

```diff
@@ -122,71 +109,80 @@ public void runValidation() {
-          if (tableConfig == null) {
-            LOGGER.warn("Table config not found for table: {}. Skipping validation.", tableNameWithType);
-            continue;
-          }
-          updateRealtimeDocumentCount(propertyStore, tableConfig);
+          updateRealtimeDocumentCount(tableConfig);
           _llcRealtimeSegmentManager.validateLLCSegments(tableConfig);
         }
       } catch (Exception e) {
-        LOGGER.warn("Exception validating table: {}", tableNameWithType, e);
+        LOGGER.warn("Caught exception while validating table: {}", tableNameWithType, e);
       }
     }
     LOGGER.info("Validation completed");
   }
 
-  private void updateRealtimeDocumentCount(ZkHelixPropertyStore<ZNRecord> propertystore, TableConfig tableConfig) {
-    final String tableNameWithType = tableConfig.getTableName();
+  private void updateRealtimeDocumentCount(TableConfig tableConfig) {
+    String realtimeTableName = tableConfig.getTableName();
     List<RealtimeSegmentZKMetadata> metadataList =
-            ZKMetadataProvider.getRealtimeSegmentZKMetadataListForTable(propertystore, tableNameWithType);
+        _pinotHelixResourceManager.getRealtimeSegmentMetadata(realtimeTableName);
     boolean countHLCSegments = true;  // false if this table has ONLY LLC segments (i.e. fully migrated)
     StreamConfig streamConfig = new StreamConfig(tableConfig.getIndexingConfig().getStreamConfigs());
     if (streamConfig.hasSimpleKafkaConsumerType() && !streamConfig.hasHighLevelKafkaConsumerType()) {
       countHLCSegments = false;
     }
     // Update the gauge to contain the total document count in the segments
     _validationMetrics.updateTotalDocumentCountGauge(tableConfig.getTableName(),
-          computeRealtimeTotalDocumentInSegments(metadataList, countHLCSegments));
+        computeRealtimeTotalDocumentInSegments(metadataList, countHLCSegments));
   }
 
   // For offline segment pushes, validate that there are no missing segments, and update metrics
-  private void validateOfflineSegmentPush(ZkHelixPropertyStore<ZNRecord> propertyStore, String offlineTableName) {
+  private void validateOfflineSegmentPush(TableConfig tableConfig) {
+    String offlineTableName = tableConfig.getTableName();
     List<OfflineSegmentZKMetadata> offlineSegmentZKMetadataList =
-        ZKMetadataProvider.getOfflineSegmentZKMetadataListForTable(propertyStore, offlineTableName);
+        _pinotHelixResourceManager.getOfflineSegmentMetadata(offlineTableName);
 
-    // Compute the missing segments if there are at least two
+    // Compute the missing segments if there are at least two segments
     int numMissingSegments = 0;
     int numSegments = offlineSegmentZKMetadataList.size();
-    if (numSegments >= 2) {
+    SegmentsValidationAndRetentionConfig validationConfig = tableConfig.getValidationConfig();
+    if (numSegments >= 2 && StringUtils.isNotEmpty(validationConfig.getTimeColumnName())) {
```

Comment:
Why do we need to import `StringUtils`?

Reply:
It's a null + empty check for simplicity